### DISCUSSION TITLE
Fastnum decimals integration

### DISCUSSION THREAD

**Discussion by neogenie** [diesel-rs/diesel#4406] [Ideas]
I've [announced](https://www.reddit.com/r/rust/comments/1hk61om/announcing_a_new_fast_exact_precision_decimal/) a new fast, exact precision decimal numbers crate [`fastnum`](https://docs.rs/fastnum/latest/fastnum/).

The current integration with Diesel is implemented in the `fastnum` crate itself under the feature flag `diesel` and
works fine.
However, I would like to move it inside Diesel. Can I expect on your help in PR review  and
reserved feature flag in diesel à la `fastnum`?

@weiznich

**Comment by weiznich** [comment]
Congratulations for the release. The crate looks really good.

Thanks for reaching out about how to handle integration with diesel. Usually the we provide integrations with other crates in diesel itself if:

* They are popular (hard for a new crate)
* The mapping to and from database types is complete (there are no values that cannot be mapped in either direction)
* The crate is reasonably stable, as it's exposed as public dependency which makes it nearly impossible to support newer crate versions if they break the API used for integration in diesel.

I'm not sure in how far the first and third point will be fulfilled in the future by fastnum. Maybe it's reasonable to first start with an out of crate integration and only later move that into diesel?
